src/wallet/wallet.h

```diff
@@ -477,16 +477,37 @@ class CWalletTx
         fInMempool = false;
         nChangeCached = 0;
         nOrderPos = -1;
+        tx_state.hashBlock = uint256();
```

Comment:
Well strictly speaking, given that Init set to unconfirmed, it would be more wrongly updating to confirmed. But if hash is set IMO we have database corruption or bug in higher tracking logic. Updated serialization logic with this and beneath to make it more robust.